fe/model.py

```diff
@@ -104,19 +209,66 @@ def predict(self, ff_params: list, mol_a: Chem.Mol, mol_b: Chem.Mol, core: np.nd
                 seed
             )
 
+            # dynamically detect if we need derivatives or not.
+            assert np.all([isinstance(x, jax.interpreters.ad.JVPTracer) for x in sys_params])
```

Comment:
Do we need this line and the loop on 217? Does this not imply the second unless sys_params is empty??

Reply:
yeah this is buggy - i removed this assertion. I'm going to rework this logic a little more - I'm not super happy with run-time reflection to see if we need to compute derivatives or not.